android/app/src/main/java/app/organicmaps/routing/RoutingErrorDialogFragment.java

```diff
@@ -55,8 +55,10 @@ private View addMessage(View frame)
   @Override
   public void onDismiss(DialogInterface dialog)
   {
-    if (mNeedMoreMaps && mCancelled)
+    if (mNeedMoreMaps && mCancelled) {
       mCancelled = false;
+      RoutingController.get().cancel();
```

Comment:
This closes the Routing menu altogether.  It is simple, yet effective.  Perhaps it would be better to add another dialog to warn the user navigation would be cancelled.